src/GovUk.Education.ExploreEducationStatistics.Common/Utils/CsvUtils.cs

```diff
@@ -83,45 +88,88 @@ public static async Task<List<string>> GetCsvHeaders(Func<Task<Stream>> streamPr
-    /// The index is the zero-based index of the row in the CSV, minus 1 if the CSV header was skipped.
+    /// The index is the zero-based index of the data row in the CSV, not including the header. Therefore the first
+    /// row of data under the header will have the index of "0".
     /// </param>
-    /// <param name="skipHeaderRow">Choose whether or not to skip a first header row in the executions.</param>
+    /// <param name="startingRowIndex">Optional parameter to skip a number of rows to be iterated over. For instance,
+    /// if this is set to 0, the first row being iterated over will be the first line of data, whereas if this is set
+    /// to 1, the first row being iterated over will be the 2nd line of data.</param>
     public static async Task ForEachRow(
         Func<Task<Stream>> streamProvider,
-        Func<List<string>, int, Task<bool>> func,
-        bool skipHeaderRow = true)
+        Func<List<string>, int, bool, Task<bool>> func,
+        int startingRowIndex = 0)
     {
-        var config = new CsvConfiguration(CultureInfo.InvariantCulture)
-        {
-            // Skip the first record of the CSV when attaching the CsvDataReader to the
-            // CsvReader if we want to omit the header row from the rows being iterated over.
-            HasHeaderRecord = skipHeaderRow
-        };
+        var config = new CsvConfiguration(CultureInfo.InvariantCulture);
 
         using var dataFileReader = new StreamReader(await streamProvider.Invoke());
         using var csvReader = new CsvReader(dataFileReader, config);
         using var csvDataReader = new CsvDataReader(csvReader);
+        var lastLine = !await csvReader.ReadAsync();
 
-        while (await csvReader.ReadAsync())
+        while (!lastLine)
         {
+            var currentRowIndex = csvReader.Parser.Row - 2;
+
+            if (currentRowIndex < startingRowIndex)
+            {
+                // Skipping this row as it is below startingRowIndex
+                lastLine = !await csvReader.ReadAsync();
+                continue;
+            }
+            
             var cellCount = csvDataReader.FieldCount;
 
             var cells = Enumerable
                 .Range(0, cellCount)
                 .Select(csvReader.GetField<string>)
                 .OfType<string>()
                 .ToList();
-
-            var currentRowIndex = csvReader.Parser.Row + (skipHeaderRow ? -1 : 0);
-
-            var result = await func.Invoke(cells, currentRowIndex);
+            
+            lastLine = !await csvReader.ReadAsync();
+            
+            var result = await func.Invoke(cells, currentRowIndex, lastLine);
 
             if (!result)
             {
                 break;
             }
         }
     }
+    
+    public static async Task Batch(
+        Func<Task<Stream>> streamProvider,
+        int batchSize,
+        Func<List<List<string>>, int, Task<bool>> func,
+        int startingBatchIndex = 0)
+    {
```

Comment:
I think this could benefit from some method comments, to explain the purpose of `func`